src/Avalonia.X11/X11Window.cs

```diff
@@ -773,21 +773,19 @@ void Cleanup()
 
         bool ActivateTransientChildIfNeeded()
         {
-            if (_transientChildren.Count == 0)
+            if (_disabled)
+            {
+                GotInputWhenDisabled?.Invoke();
                 return false;
-            var child = _transientChildren.First();
-            if (!child.ActivateTransientChildIfNeeded())
-                child.Activate();
+            }
+
             return true;
         }
-        
+
         void SetTransientParent(X11Window window, bool informServer = true)
```

Comment:
`Show` resets the parent in current master.  This method should be removed and `SetTransientForHint` should be renamed to `SetParent` and not called from X11Window.cs itself at all.

Reply:
would you mind sorting this, im not that comfortable with x11 yet!

Reply:
@kekekeks sorted it.